bulk_deploy/tsuru_api_client.rb

```diff
@@ -0,0 +1,438 @@
+require "net/http"
+require "uri"
+require "cgi"
+require "json"
+require "yajl"
+require "openssl"
+
+class TsuruAPIClient
+
+  # Default headers added to each request
+  HEADERS = {
+    "Content-Type"    => "application/json",
+    "Accept-Encoding" => "gzip",
+  }
+
+  VERB_MAP = {
+    :get    => Net::HTTP::Get,
+    :post   => Net::HTTP::Post,
+    :put    => Net::HTTP::Put,
+    :delete => Net::HTTP::Delete
+  }
+
+  def initialize(logger:, environment:, host:, protocol: "https://")
+    # Parse the Tsuru API domain and create a HTTP client
+    uri = URI.parse(protocol + environment + "-api." + host)
+    @http = Net::HTTP.new(uri.host, uri.port)
+
+    # Set SSL flag to true if protocol is HTTPS
+    if protocol == "https://"
+      @http.use_ssl = true
+      @http.verify_mode = OpenSSL::SSL::VERIFY_NONE
```

Comment:
The API should have a valid certificate. I'd prefer to use a different CA bundle (like alphagov/tsuru-ansible@3878b057f613b0a59c70a433598fdbb147939356) rather than disable verification altogether.